appcat/objectbucket/composition.yaml

```
@@ -0,0 +1,88 @@
+---
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: default-objectbucket-cloudscale
+spec:
+  compositeTypeRef:
+    apiVersion: appcat.vshn.io/v1
+    kind: XObjectBucket
+  writeConnectionSecretsToNamespace: crossplane-system
+  patchSets:
+    - name: annotations
+      patches:
+      - type: FromCompositeFieldPath
+        fromFieldPath: metadata.annotations
+    - name: labels
+      patches:
+      - type: FromCompositeFieldPath
+        fromFieldPath: metadata.labels
+    - name: namespace
+      patches:
+      - type: FromCompositeFieldPath
+        fromFieldPath: metadata.labels[crossplane.io/claim-namespace]
+        toFieldPath: spec.forProvider.manifest.metadata.namespace
+  resources:
+    - base:
+        apiVersion: cloudscale.crossplane.io/v1
+        kind: Bucket
+        metadata:
+          name: ""
+        spec:
+          forProvider:
+            bucketName: ""
+            credentialsSecretRef:
+              name: ""
+              namespace: provider-cloudscale-system
+            endpointURL: ""
+            region: ""
+      patches:
+        - type: PatchSet
+          patchSetName: annotations
+        - type: PatchSet
+          patchSetName: labels
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.parameters.bucketName
+          toFieldPath: metadata.name
```

I wouldn't set the metadata name the same as the bucketname, since users could potentially name the same bucket twice. (ofc 2 same-named buckets wouldn't work in cloudscale, but at least cloudscale would report the error and not at Kubernetes level)
```suggestion
          fromFieldPath: metadata.labels[crossplane.io/composite]
          toFieldPath: metadata.name
```